src/trim_comparison.rs

```diff
@@ -0,0 +1,808 @@
+                    "Same({2})",
+                    "Same({2})",
+                    "Same({2})",
+                    "Same({2})",
+                    "Same({2})",
+                    "Same({2})",
+                    "Same({2})",
+                    "Same({2})",
+                    "Same({2})",
+                    "Same({2})",
+                    "Same({2})",
+                    "Same({2})",
+                    "Same({2})",
+                    "Same({2})",
+                    "Same({2})",
+                    "Same({2})",
+                    "Same({2})",
+                    "Same({2})",
+                    "Same({2})",
+                    "Same({2})",
+                    {
+                      "address": "Same(0x49d36570d4e46f48e99674bd3fcc84644ddd6b96f7c741b1562b82f9e004dc7)",
+                      "storage_entries": [
+                        {
+                          "key": "Same(0x5496768776e3db30053404f18067d81a6e06f5a2b0de326e21298fd9d569a9a)",
+                          "value": {
+                            "Different": {
+                              "base": "0x20c632e33333b4c5",
+                              "native": "0x20c62db4af3ea769"
+                            }
+                          }
+                        },
+                      ]
+                    },
+                  ]
+                },
+              },
+              "transaction_hash": "0x3ae644195902851108e62c261c27cf74018af096404b6799053494727faf94b"
+            },
+          ]
+        });
+    }
+
+    #[test]
+    fn test_over_10() {
+        let obj = json!({
+          "root" : [
+            "Same({0})",
+            "Same({1})",
+            "Same({2})",
+            "Same({3})",
+            "Same({4})",
+            "Same({5})",
+            "Same({6})",
+            "Same({7})",
+            {
+              "key": "Same(0x5496768776e3db30053404f18067d81a6e06f5a2b0de326e21298fd9d569a9a)",
+              "value": {
+                "Different": {
+                  "base": "0x20c632e33333b4c5",
+                  "native": "0x20c62db4af3ea769"
+                }
+              }
+            },
+            "Same({9})",
+            "Same({10})",
+            "Same({11})",
+            "Same({12})",
+            "Same({13})",
+            "Same({14})",
+          ]
+        });
+
+        let expected_trimmed_obj = json!({
+          "root" : [
+            {
+              "key": "Same(0x5496768776e3db30053404f18067d81a6e06f5a2b0de326e21298fd9d569a9a)",
+              "value": {
+                "Different": {
+                  "base": "0x20c632e33333b4c5",
+                  "native": "0x20c62db4af3ea769"
+                }
+              }
+            },
+          ]
+        });
+
+        let config = TrimConfig { level: 1 };
+        let trimmed_obj = trim_comparison(obj, &config);
+        assert_eq!(trimmed_obj, expected_trimmed_obj);
+    }
+
+    #[test]
+    fn test_trim() {
```

Comment:
Reduce a bit the size of this test

Reply:
I broke up `test_trim` into `test_realistic_example_` with different levels to test. I also, changed the keys to be easier to read and removed some of the fields without taking away from the test.